### PR TITLE
Add instructions for replicating functionality in SpikeInterface

### PR DESCRIPTION
**Pull request by jsiegle**
The Allen Institute is in the process of migrating its spike sorting pipelines to SpikeInterface, with no plans to support `ecephys_spike_sorting` moving forward.

These changes add a recommendation for new and existing users to base their spike sorting pipelines on SpikeInterface. They also add code snippets to `README` files that demonstrate how to replicate the functionality of `ecephys_spike_sorting` in SpikeInterface.

### CODE REVIEW

**Comment by jsiegle** [comment]
@alejoe91 can you review the SpikeInterface examples?

File: ecephys_spike_sorting/modules/automerging/README.md [hunk 1]
```diff
@@ -1,12 +1,39 @@
...
 
-This is not currently part of our pipeline since switching to Kilosort2, but we're keeping the code around in case others find it useful. For example, it could be helpful for matching units across a series of chronic recordings.
+Searches for clusters that likely belong to the same cell, and merges them automatically.
 
+This module has not been used since switching to Kilosort 2, which has far fewer split units than Kilosort 1. We're keeping the code around in case others find it useful. For example, it could be helpful for matching units across a series of chronic recordings. However, much more complete implementations of this functionality exist elsewhere ([UnitMatch](https://github.com/EnnyvanBeest/UnitMatch), for example).
+
+
+### SpikeInterface implementation
+
+SpikeInterface does not currently include the ability to automatically merge units, but this is under active development. Information that is helpful for making merge decisions, such as waveform similarity and cross-correlograms, can be computed using the `postprocessing` module:
```
Comment (alejoe91) [inline ecephys_spike_sorting/modules/automerging/README.md:10]: @jsiegle we have something similar that might be worth mentioning: https://spikeinterface.readthedocs.io/en/latest/modules/curation.html#automatic-curation-tools

In particular: https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.curation.get_potential_auto_merge

The `get_potential_auto_merge()` function basically looks for potential merges mased on template similarity, cross-correlgrams, and improved ISI violation quality

File: ecephys_spike_sorting/modules/kilosort_postprocessing/README.md [hunk 2]
```diff
@@ -8,8 +8,35 @@ This module aims to correct for this by removing spikes from the same unit or ne
 
 We are not currently taking into account spike amplitude when removing spikes; the module just deletes one spike from an overlapping pair that occurs later in time.
 
-Running
--------
+### SpikeInterface implementation
+
+There is not currently a function for removing putative double-counted spikes with SpikeInterface. Instead, you can use the `export_to_phy()` method to save the data in a format that can be loaded by this module:
```
Comment (alejoe91) [inline ecephys_spike_sorting/modules/kilosort_postprocessing/README.md:13]: We do have a `remove_duplicated_spikes` function:
https://spikeinterface.readthedocs.io/en/latest/api.html#spikeinterface.curation.remove_duplicated_spikes

File: ecephys_spike_sorting/modules/mean_waveforms/README.md [hunk 3]
```diff
@@ -20,9 +20,30 @@ Metrics are computed for every waveform, and include features of the 1D peak-cha
...
+```python
+import spikeinterface.full as si
+
+from spikeinterface.postprocessing import compute_template_metrics
+
+waveform_extractor = si.extract_waveforms(recording=recording, 
+                                          sorting=sorting, 
+                                          folder='waveforms')
+
+_ = compute_template_metrics(waveform_extractor)
```
Comment (alejoe91) [inline ecephys_spike_sorting/modules/mean_waveforms/README.md:38]:
```suggestion
tm = compute_template_metrics(waveform_extractor)
display(tm)
```
shows that `tm` is a `pandas.DataFrame`

File: ecephys_spike_sorting/modules/median_subtraction/README.md [hunk 4]
```diff
@@ -1,15 +1,55 @@
...
+raw_rec = si.read_spikeglx('/path/to/data', 
+                           stream_name='imec0.ap',     
+                           load_sync_channel=False)
+
+# perform high-pass filtering (especially important for Neuropixels 2.0 data):
+rec_filt = si.highpass_filter(raw_rec, 
+                              freq_min=400.)
+
+# remove channels outside the brain
+bad_channel_ids, channel_labels = si.detect_bad_channels(rec_filt)
```
Comment (alejoe91) [inline ecephys_spike_sorting/modules/median_subtraction/README.md:31]: note that this removes also noisy and dead channels, not just out of brain

**Review by alejoe91 — COMMENTED** [review]
Looking great @jsiegle 

Left a few comments and suggestions!

**Comment by jsiegle** [comment]
Awesome, thanks a lot!

**Pull request merged**